tests/specifications/e2e/pipeline_hierarchical.yaml

```diff
@@ -0,0 +1,17 @@
+step_1:
```

Comment:
Cool, I do think this is important. In my head, we'll only ever have one e2e test and that pipeline will just include all types of steps (hierarchical, looping, parallel, whatever else). And we can address what to do if it starts taking too dang long if/when that happens